src/map/mob_spell_container.cpp

```diff
@@ -85,6 +91,52 @@ void CMobSpellContainer::AddSpell(SpellID spellId)
     }
 }
 
+void CMobSpellContainer::RemoveSpell(SpellID spellId)
+{
+    auto findAndRemove = [](std::vector<SpellID>& list, SpellID id)
+    {
+        list.erase(std::remove(list.begin(), list.end(), id), list.end());
+    };
+
+    findAndRemove(m_gaList, spellId);
+    findAndRemove(m_damageList, spellId);
+    findAndRemove(m_buffList, spellId);
+    findAndRemove(m_debuffList, spellId);
+    findAndRemove(m_healList, spellId);
+    findAndRemove(m_naList, spellId);
+
+    m_hasSpells = !(m_gaList.empty() && m_damageList.empty() && m_buffList.empty() && m_debuffList.empty() && m_healList.empty() && m_naList.empty());
+}
+
+std::optional<SpellID> CMobSpellContainer::GetBestAvailable(SPELLFAMILY family)
+{
+    std::vector<SpellID> matches;
+    auto searchInList = [&](std::vector<SpellID>& list)
+    {
+        for (auto id : list)
+        {
+            auto spell = spell::GetSpell(id);
+            bool sameFamily = (family == SPELLFAMILY_NONE) ? true : spell->getSpellFamily() == family;
+            bool hasEnougnMP = spell->getMPCost() <= m_PMob->health.mp;
+            bool isNotInRecast = !m_PMob->PRecastContainer->Has(RECAST_MAGIC, static_cast<uint16>(id));
+            if (sameFamily && hasEnougnMP && isNotInRecast)
+            {
+                matches.push_back(id);
+            }
+        };
+    };
+    searchInList(m_gaList);
+    searchInList(m_damageList);
+    searchInList(m_buffList);
+    searchInList(m_debuffList);
+    searchInList(m_healList);
+    searchInList(m_naList);
```

Comment:
"Hey in the future we should only search lists that would have the appropriate spell."
"I know; that's why I want to give spell families parent families."